core/src/com/unciv/ui/worldscreen/WorldScreenTopBar.kt

```diff
@@ -158,9 +158,13 @@ class WorldScreenTopBar(val worldScreen: WorldScreen) : Table() {
         }
 
         // in case the icons are configured higher than a label, we add a dummy - height will be measured once before it's updated
-        resourcesWrapper.add(resourceActors[0].icon)
+        if (resourceActors.isNotEmpty()) {
+            resourcesWrapper.add(resourceActors[0].icon)
```

Comment:
If resource wrapper is only used in this 'if' case, we should move all the initialization if this object to within the if

Reply:
Not Skekdog's fault. I probably expected more cross-method reuse.

Reply:
I've updated WorldScreenTopBar to include this change
resourceTable still has to be initialized because of return resourceTable, and I kept in the resourceTable padding because it looked better and allowed the turnsLabel to not overlap with the civ name